Devicedata

Krishnan-Priyanshu opened the pull request:
# Description
Add the function to get all the devices and filter the device based on the field from the system struct

# GitHub Issues
List the GitHub issues impacted by this PR:
https://github.com/dell/csm/issues/778

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Integration Tests

![image](https://github.com/dell/goscaleio/assets/118425422/53ff5776-8213-4937-8b7e-cf84031ec0a6)

VamsiSiddu-7 reviewed (commented):
@Krishnan-Priyanshu Changes Looks good to me. Requesting you to correct the indentation of the github issue and provide the tests results. Thanks

Krishnan-Priyanshu commented:
> @Krishnan-Priyanshu Changes Looks good to me. Requesting you to correct the indentation of the github issue and provide the tests results. Thanks

Thanks @VamsiSiddu-7 for the review.
we've fixed the GitHub issue indentation and added the test results.
